frontend/components/Layout/Layout.tsx

```diff
@@ -1,18 +1,24 @@
 import { useEffect } from "react";
+import { ToastContainer } from "react-toastify";
 import { useRouter } from "next/router";
 import { useSession, signOut } from "next-auth/react";
 import { styled } from "../../stitches.config";
-import { ShouldRenderNav } from "../../utils/PagesNames";
 import NavBar from "../NavBar/NavBar";
 import Flex from "../Primitives/Flex";
 import { REFRESH_TOKEN_ERROR } from "../../utils/constants";
+import "react-toastify/dist/ReactToastify.css";
+import { ShouldRenderNav } from "../../utils/routes";
+import { instance } from "../../utils/fetchData";
 
 const Main = styled("main", Flex, { px: "3vw", py: "$50" });
 
 const Layout: React.FC = ({ children }) => {
   const router = useRouter();
   const { data: session, status } = useSession({ required: false });
 
+  if (status === "authenticated")
+    instance.defaults.headers.common.Authorization = `Bearer ${session?.accessToken}`;
```

Comment:
This can't be here, it's being executed on every rerender. 
It needs to be on a `useEffect` at least.

But I don't think it's the right place to set the token, weren't you able to find a place after login?

```suggestion
useEffect(() => {
  if (status === "authenticated" && session?.accessToken) {
    instance.defaults.headers.common.Authorization = `Bearer ${session?.accessToken}`;
  }
}, [status])
```

Reply:
Thanks bro, I just fixed that